src/ray/raylet/node_manager.h

```diff
@@ -320,11 +320,11 @@ class NodeManager : public rpc::NodeManagerServiceHandler {
   ///
   /// \param actor_id The actor ID of the actor whose state was updated.
   /// \param data Data to publish.
-  /// \param failure_callback An optional callback to call if the publish is
+  /// \param tolerate_failure If tolerate update failure, if not will call Log Fatal
```

Comment:
Do we need `tolerate_failure` according to current code base?  I think it's better to introduce this in `high-available GCS` PR.

Reply:
> Do we need `tolerate_failure` according to current code base? I think it's better to introduce this in `high-available GCS` PR.

I do think GCS Client return status is OK. Seems NodeManager need make the choice if "tolerate_failure" according to current code base. But in future it might either retry in GCS Client or retry in NodeManager.

Reply:
We're not handling the network error nor gcs unavailable error here.
In this case, when a node dies, all other nodes will simultaneously try to mark the actors on that node as dead or reconstructing. The `log_length` is used to make sure only one node can successfully updated the actor state. 
So `tolerate_failure` isn't a proper name here. And that's also why I suggested hiding this logic inside a method.

Reply:
see .cc please, only use `tolerate_failure` to compatible with the original logic and remove redundant code

Reply:
I know you didn't change the logic. But maybe rename the parameter to something like `concurrent_update` to avoid confusion